### PR TITLE
Documentation consistency improvements

### PR DESCRIPTION
**Pull request by eliasp**
### What does this PR do?
It cleans up many smaller inconsistencies in the docs, such as:
- lack of rST verbatim usage (like `code` in Markdown)
- lack of rST references (e.g. when referencing another module/function)
- proper usage of field lists for function parameters/return data, etc.
- inconsistent module names in module listings
- ...

Each logical change is a separate commit.

### What issues does this PR fix or reference?
None

### Merge requirements satisfied?
- [x] Docs
- [x] (N/A) Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] (N/A) Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by eliasp** [comment]
Don't have time right now to add further improvments, so I'm switching from WIP to "Ready for Review" and also just rebased it on latest master.

**Review by Ch3LL — DISMISSED** [review]

**Comment by eliasp** [comment]
Rebased it on latest saltstack:master to fix the merge conflicts.

**Review by Ch3LL — CHANGES_REQUESTED** [review]

**Review by Ch3LL — DISMISSED** [review]

**Review by ScriptAutomate — CHANGES_REQUESTED** [review]
My only feedback is with the changes to `salt/states/pkgrepo.py` when it comes to **GPG** being changed to **PGP**. Is there a good justification for why this needs to be modified in this states module? Isn't GPG an open source standard implementation of PGP, and that most of these should still be saying `GPG`? It's difficult for me to discern whether any reference to a "GPG key" is an accidental online phenomenon of people incorrectly using it interchangeably with "PGP key" or whether the _true_ definition is always a PGP key that is being mistaken. It seems that since it is a non-proprietary software implementation of the PGP standard, that GPG may make sense to continue using?

That, or perhaps we can use `PGP/GPG` if they are meant to be interchangeable, allowing for both to appear when searching docs?

**Comment by ScriptAutomate** [comment]
Side note for those doomed to try reviewing a major amount of diffs in GitHub for a single PR similar to this PR in nature:

- Scroll down the entire list of changes so that GitHub loads all of the collapsible boxes associated with all of the changes
- Open up the inspector/console in the browser devtools. This is probably accessed easily with `ctrl+i`. We want to be in the **Console**, which in Firefox is `ctrl+shift+k`.
- Type in the following to expand all diffs, in order to avoid needing to click on hundreds of **Load diff** buttons:
```js
Array.from(document.getElementsByClassName('load-diff-button')).map(button => button.click())
```

Wish GitHub had a _Load all diffs_ button. This took a lot of time to eyeball.

Source GitHub conversation where I found the snippet: https://github.com/sindresorhus/refined-github/issues/825#issuecomment-593112737

File: doc/ref/beacons/all/salt.beacons.log_beacon.rst [hunk 1]
```diff
@@ -1,6 +1,6 @@
-==============================
-salt.beacons.log_beacon module
-==============================
+salt.beacons.log_beacon
+=======================
+>>>>>>> doc: remove `module` suffix from descriptions
```
[the remark this answers is not in the document]
Comment (eliasp) [reply doc/ref/beacons/all/salt.beacons.log_beacon.rst]: Thanks for fixing this! Don't know how this slipped through...

**Comment by eliasp** [comment]
> That, or perhaps we can use `PGP/GPG` if they are meant to be interchangeable, allowing for both to appear when searching docs?

- correctness: `PGP`
- searchability: `PGP/GPG`
- compromise: `PGP/GnuPG`

I'm undecided :)

**Comment by ScriptAutomate** [comment]
> > That, or perhaps we can use `PGP/GPG` if they are meant to be interchangeable, allowing for both to appear when searching docs?
> 
>     * correctness: `PGP`
> 
>     * searchability: `PGP/GPG`
> 
>     * compromise: `PGP/GnuPG`
> 
> 
> I'm undecided :)

I'm thinking of leaving with `gpg` for now, due to much of configs and codebase making references to gpg keys:

```bash
egrep -rn "gpgkey|gpg_key"
```

Due to that, and with other docsets such as the GitHub, GitLab, and `git` docs as examples:
- [GitHub Docs: Generating a New GPG Key](https://docs.github.com/en/github/authenticating-to-github/generating-a-new-gpg-key)
- [`git` docs: Git Tools - Signing Your Work](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work)

```
If you have a GPG private key setup, you can now use it to sign new tags. All you have to do...
```

GitLab seems to use `GPG key` but [also provides the following message about their docs](https://docs.gitlab.com/ee/user/project/repository/gpg_signed_commits/):

```
The term GPG is used for all OpenPGP/PGP/GPG related material and implementations.
```

This seems important, because keys will be having `-----BEGIN PGP SIGNATURE-----` within them, too, to your point on technical correctness.

I'm going to go ahead and apply the commit suggestions I've made for `GPG` in the docs descriptions. I'm going to also create an issue to use GitLab's approach so that we can have an `.. include::` of a reusable `.. note::` admonition about the use of GPG in reference to OpenPGP/PGP/GPG.

**Review by ScriptAutomate — APPROVED** [review]

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**